detection_rules/rule.py

```diff
@@ -267,6 +269,64 @@ def get_build_fields(self) -> {}:
         return build_fields
 
 
+@dataclass(frozen=True)
+class Osquery:
+    query: str
+    label: str
+
+    @classmethod
+    def from_string(cls, entry: str) -> 'Osquery':
+        match = re.match(r'!{osquery({.*})}', entry)
```

Comment:
How strict are the osquery and insights syntax? Can they have whitespace or linebreaks? Those would both break this pattern matching

Reply:
looks like certain line breaks are legal

![image](https://user-images.githubusercontent.com/16747370/220983029-fa3024f6-ab6e-4160-af97-3a28a411e142.png)

![image](https://user-images.githubusercontent.com/16747370/220983094-f5b21913-8c4f-4edf-a530-b6adf340f77a.png)